src/Reflex/Class.hs

```diff
@@ -791,6 +795,64 @@ zipListWithEvent f l e = do
   mapAccumMaybe_ f' l e
 
 --------------------------------------------------------------------------------
+-- AttachApply
+--------------------------------------------------------------------------------
+
+-- | An 'AttachApply' type carries a function that can be applied to an 'Event' whenever the 'Event' fires.
+--
+-- 'AttachApply' is intended to be used via the '<@>' and '<@' operators and in conjunction with the `Applicative` interface.
+--
+-- If we have:
+-- > f  :: a -> b -> c -> d
+-- > g  :: a -> b -> d
+-- > b1 :: Behavior t a
+-- > b2 :: Behavior t b
+-- > e  :: Event t c
+-- and an 'AttachApply' instance for `Behavior`, then we get:
+-- > f <$> b1 <*> b2 <@> e :: Event t d
+-- > g <$> b1 <*> b2 <@  e :: Event t d
+--
+-- It is a handy way to combine several 'Behavior's or 'Dynamic's together so that you can sample them all at the times when an `Event` is firing.
+class Reflex t => AttachApply t f where
+  attachApply :: f (a -> b) -> Event t a -> Event t b
```

Comment:
This looks good, but we might want something like `attachApplyWithMaybeM :: (a -> b -> PushM t (Maybe c)) -> f a -> Event t b -> Event t c` to be in the typeclass, since it allows a wider range of functions to be to implemented with good performance.  Then, we can just define `attachApply` as `attachApplyWithMaybeM (\f x -> return $ Just $ f x)`.